sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -119,33 +119,34 @@ case class Like(left: Expression, right: Expression) extends StringRegexExpressi
       if (rVal != null) {
         val regexStr =
           StringEscapeUtils.escapeJava(escape(rVal.asInstanceOf[UTF8String].toString()))
-        ctx.addMutableState(patternClass, pattern,
-          s"""$pattern = ${patternClass}.compile("$regexStr");""")
```

Comment:
This can avoid compiling the regex every time.

Reply:
but we are adding a new one every time, so we are compiling it every time. Am I wrong?

Reply:
no, the code generating logic runs only once, the generated code runs many times, for each input row.

Reply:
Sorry, you are right, my bad mistake. I am closing this PR immediately, sorry.

Reply:
ah that's ok, Spark's codegen framework is pretty hard to understand, it takes time to get familiar with it :)